internal/operators/ocs/validations.go

```diff
@@ -2,247 +2,251 @@ package ocs
+		o.log.Info("Empty Inventory for host with hostID ", *host.ID)
+		nodeInfo.missingInventory = true
+		status = "Missing Inventory in some of the hosts"
+		//return status, nil
+		return status, errors.New("Missing Inventory in some of the hosts ") // to indicate that inventory is empty and the ValidationStatus must be Pending
 	} else if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
 		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
-		return false, err
+		return status, err
 	}
 
 	disks := getValidDiskCount(inventory.Disks)
-
-	if disks > 1 { // OCS must use the non-boot disks
-		requiredDiskCPU := int64(disks-1) * o.OCSRequiredDiskCPUCount
-		requiredDiskRAM := int64(disks-1) * o.OCSRequiredDiskRAMGB
+	if disks > 1 { // OCS must use only the non-boot disks
+		requiredDiskCPU := int64(disks-1) * o.config.OCSRequiredDiskCPUCount
+		requiredDiskRAM := int64(disks-1) * o.config.OCSRequiredDiskRAMGB
 
 		if inventory.CPU.Count < requiredDiskCPU || inventory.Memory.UsableBytes < gbToBytes(requiredDiskRAM) {
-			status := fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy OCS. The hosts has ", disks, " disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " RAMGB")
-			*insufficientHosts = append(*insufficientHosts, status)
+			status = fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy OCS. The hosts has ", disks, " disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " RAMGB")
+			nodeInfo.insufficientHosts = append(nodeInfo.insufficientHosts, status)
 			o.log.Info(status)
 		} else {
-			*cpuCount += (inventory.CPU.Count - requiredDiskCPU)                     // cpus excluding the cpus required for disks
-			*totalRAM += (inventory.Memory.UsableBytes - gbToBytes(requiredDiskRAM)) // ram excluding the ram required for disks
-			*diskCount += (int64)(disks - 1)                                         // not counting the boot disk
-			*hostsWithDisks++
+			nodeInfo.cpuCount += inventory.CPU.Count - requiredDiskCPU                // cpus excluding the cpus required for disks
+			nodeInfo.ram += inventory.Memory.UsableBytes - gbToBytes(requiredDiskRAM) // ram excluding the ram required for disks
+			nodeInfo.numDisks += (int64)(disks - 1)                                   // not counting the boot disk
+			nodeInfo.hostsWithDisks++
 		}
 	} else {
-		*cpuCount += inventory.CPU.Count
-		*totalRAM += inventory.Memory.UsableBytes
+		nodeInfo.cpuCount += inventory.CPU.Count
+		nodeInfo.ram += inventory.Memory.UsableBytes
 	}
-	return false, nil
+	return status, nil
 }
 
-func gbToBytes(gb int64) int64 {
-	return gb * int64(units.GB)
-}
-
-// count all disks of drive type ssd or hdd
-func getValidDiskCount(disks []*models.Disk) int {
-
-	var countDisks int
+// used to validate resource requirements for OCS excluding disk requirements and set a status message
```

Comment:
Please change the comment - this function now validates the disk requirements as well.

Reply:
Would it make sense to rename this function canOCSBeDeployed ()?

Reply:
Yes that would make sense